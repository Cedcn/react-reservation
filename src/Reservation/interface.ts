import { Moment } from 'moment'
import { isArray } from 'lodash'

export type CalendarValue = Moment | null
export type WeekCode = 0 | 1 | 2 | 3 | 4 | 5 | 6

export type Theme = any

export interface RepeatDaysByWeek {
  availableWeeks?: WeekCode[]
  startDay?: Moment
  endDay?: Moment
}

export type FixedDays = Moment[]

export type Days = FixedDays | RepeatDaysByWeek

export interface Calendar {
  prefixCls?: string
  value: CalendarValue
  onChange: (value: CalendarValue) => void
  days: Days
}

export interface CalendarTableCommonProps {
  currentMonthIdx: number
  prefixCls: string
  startDay: Moment
  endDay?: Moment
  value: CalendarValue
  availableWeeks?: WeekCode[]
  availableDays?: FixedDays
  onChange: (value: CalendarValue) => void
  toNext: () => boolean
  toLast: () => boolean
}

export interface CellStatus {
  isMakefull?: boolean
  isSelectable?: boolean
  isBeforeStartDayMinute?: boolean
  isAfterEndDayMinute?: boolean
  isBeforeStartDay?: boolean
  isAfterEndDay?: boolean
  isSelected?: boolean
  isNotChecked?: boolean
  isToday?: boolean
  isStartDate?: boolean
  isEndDate?: boolean
  isLastMonthDay?: boolean
  isNextMonthDay?: boolean
}

export function isFixedDays(days: Days): days is FixedDays {
  return isArray(days)
}
