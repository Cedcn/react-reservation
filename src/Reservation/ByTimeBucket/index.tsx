/** @jsx jsx */
import { jsx } from '@emotion/core'
import React from 'react'
import moment, { Moment } from 'moment'
import TimeBucketTable from './TimeBucketTable'
import TimeBucketTabs from './TimeBucketTabs'
import { map } from 'lodash'
import {
  isSpecifiedDays,
  Days,
  TimeBucketValue,
  SectionRanges,
  Offset,
  ByTimeBucketCellProps,
  isSameSectionRanges,
  ByTimeBucketQuota,
} from '../interface'

import { getNow, getDateByArea } from '../utils'
import * as styles from './styles'

interface TimeBucketProps {
  prefixCls?: string
  days?: Days
  value?: TimeBucketValue
  onChange: (value?: TimeBucketValue) => void
  mode?: 'tabs' | 'table'
  ranges: SectionRanges
  advance?: Offset | boolean
  isMultiple?: boolean
  area?: Offset
  isMinShort?: boolean
  quotaRequest?: (start: Moment, end: Moment) => Promise<ByTimeBucketQuota[]>
  cellRenderer?: React.ComponentType<ByTimeBucketCellProps>
}

const TimeBucket: React.FC<TimeBucketProps> = (props) => {
  const {
    prefixCls = 'rTb',
    days,
    value,
    onChange,
    ranges,
    mode = 'table',
    advance,
    isMultiple,
    isMinShort,
    area,
    cellRenderer,
    quotaRequest,
  } = props

  let startDay: Moment | null | undefined
  let endDay: Moment | null | undefined
  let disabledWeeks
  let disabledDays
  let specifiedDays

  if (isSpecifiedDays(days)) {
    startDay = moment.min(days)
    endDay = moment.max(days)
    specifiedDays = map(days, (day) => day.clone().startOf('day'))
  } else {
    startDay = days?.startDay
    endDay = days?.endDay
    disabledWeeks = days?.disabledWeeks
    disabledDays = days?.disabledDays
  }
  const today = getNow()
  if (!startDay || (startDay && startDay.isBefore(today, 'minute'))) {
    startDay = today
  }

  if (area) {
    const endOffsetDay = getDateByArea(area)

    if (!endDay || (endDay && endDay.isAfter(endOffsetDay, 'minute'))) {
      endDay = endOffsetDay
    }
  }

  const commonProps = {
    prefixCls,
    startDay,
    endDay,
    disabledWeeks,
    disabledDays,
    specifiedDays,
    value,
    onChange,
    advance,
    isMultiple,
    isMinShort,
    cellRenderer,
    quotaRequest,
  }

  let content

  if (mode === 'tabs') {
    content = <TimeBucketTabs ranges={ranges} {...commonProps} />
  }

  if (mode === 'table') {
    content = isSameSectionRanges(ranges) ? (
      <TimeBucketTable ranges={ranges} {...commonProps} />
    ) : (
      <TimeBucketTabs ranges={ranges} {...commonProps} />
    )
  }

  return (
    <div className={`${prefixCls}-time-bucket`} css={styles.byTimeBucket}>
      {content}
    </div>
  )
}

export default TimeBucket
