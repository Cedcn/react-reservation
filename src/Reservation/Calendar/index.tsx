/** @jsx jsx */
import { jsx } from '@emotion/core'
import React, { useState } from 'react'
import moment from 'moment'
import CalendarHeader from './CalendarHeader'
import CalendarTable from './CalendarTable'
import { Calendar, isFixedDays } from '../interface'
import styles from '../styles'

interface CalendarProps extends Calendar {}

const ReservationCalendar: React.FC<CalendarProps> = (props) => {
  let canToNext = true
  let canToLast = true
  const [currentMonthIdx, setCurrentMonthIdx] = useState(0)

  const toNextMonth = () => {
    if (!canToNext) return false
    setCurrentMonthIdx(currentMonthIdx + 1)
    return true
  }

  const toLastMonth = () => {
    if (!canToLast) return false
    setCurrentMonthIdx(currentMonthIdx - 1)
    return true
  }

  const { prefixCls = 'rV', days, value } = props

  const today = moment()

  if (!isFixedDays(days)) {
    const { startDay, endDay, availableWeeks } = days

    const startMonthDay = startDay && startDay.isAfter(today, 'minute') ? startDay.startOf('day') : today.startOf('day')
    const endMonthDay = endDay ? endDay.endOf('day') : undefined
    const onChange = () => {}
    const currentMonthDay = startMonthDay.clone().month(currentMonthIdx + startMonthDay.month())

    canToLast = startMonthDay.isBefore(currentMonthDay, 'month')
    canToNext = !endMonthDay || (endMonthDay && endMonthDay.isAfter(currentMonthDay, 'month'))

    const commonProps = {
      prefixCls,
      currentMonthIdx,
      startDay: startMonthDay,
      endDay: endMonthDay,
      availableWeeks,
      value,
      onChange,
      toLast: toLastMonth,
      toNext: toNextMonth,
    }

    return (
      <div className={prefixCls} css={styles.reservation}>
        <CalendarHeader
          prefixCls={prefixCls}
          currentDay={currentMonthDay}
          canToLast={canToLast}
          canToNext={canToNext}
          toNext={toNextMonth}
          toLast={toLastMonth}
        />
        <CalendarTable {...commonProps} />
      </div>
    )
  }

  return null
}

export default ReservationCalendar
