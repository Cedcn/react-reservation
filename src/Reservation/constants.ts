export const DATE_ROW_COUNT = 6
export const DATE_COL_COUNT = 7
export const MAX_SHOW_QUOTA = 9999
export const WEEK_CODES = [0, 1, 2, 3, 4, 5, 6]
