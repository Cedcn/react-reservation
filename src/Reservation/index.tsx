/** @jsx jsx */
import { jsx } from '@emotion/core'
import { ThemeProvider } from 'emotion-theming'
import React, { useState } from 'react'
import ReservationCalendar from './Calendar'
import { CalendarValue, TimeBucketsValue } from './interface'
import moment from 'moment'
import ReservationTimeBuckets from './TimeBuckets'

const Reservation: React.FC<any> = (props) => {
  const theme = { borderColor: '#eee' }
  const [value, setValue] = useState<CalendarValue>(null)
  const onChange = (value: CalendarValue) => {
    setValue(value)
  }

  const [timeBucketsValue, setTimeBucketsValue] = useState<TimeBucketsValue>(null)
  const onTimeBucketsChange = (value: TimeBucketsValue) => {
    setTimeBucketsValue(value)
  }

  return (
    <ThemeProvider theme={theme}>
      <h2>Repeat </h2>
      <ReservationCalendar value={value} onChange={onChange} />
      <h2>Repeat, set disabled weeks and set disabled days</h2>
      <ReservationCalendar
        value={value}
        onChange={onChange}
        days={{ disabledWeeks: [0, 1], disabledDays: [moment('2020-04-03')] }}
      />
      <h2>Repeat, set start day and set end day</h2>
      <ReservationCalendar
        value={value}
        onChange={onChange}
        days={{ startDay: moment('2020-02-03'), endDay: moment('2020-05-03') }}
      />
      <h2>Specified days</h2>
      <ReservationCalendar value={value} onChange={onChange} days={[moment('2020-04-03'), moment('2020-02-04')]} />
      <h2>Time Buckets</h2>
      <ReservationTimeBuckets
        value={timeBucketsValue}
        onChange={onTimeBucketsChange}
        days={{ startDay: moment('2020-02-03'), endDay: moment('2020-05-03') }}
        ranges={[
          { start: [10, 10], end: [11, 20] },
          { start: [11, 30], end: [13, 30] },
          { start: [13, 30], end: [15, 30] },
          { start: [15, 30], end: [20, 30] },
        ]}
      />
      <h2>Time Buckets list</h2>
      <ReservationTimeBuckets
        value={timeBucketsValue}
        onChange={onTimeBucketsChange}
        days={{ startDay: moment('2020-02-03'), endDay: moment('2020-05-03') }}
        ranges={[
          { start: [10, 10], end: [11, 20] },
          { start: [11, 30], end: [13, 30] },
          { start: [13, 30], end: [15, 30] },
          { start: [15, 30], end: [20, 30] },
        ]}
        mode="tabs"
      />
    </ThemeProvider>
  )
}

export default Reservation
