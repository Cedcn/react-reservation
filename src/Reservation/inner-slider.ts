export const getSwipeDirection = (touchObject: any) => {
  var xDist, yDist, r, swipeAngle
  xDist = touchObject.startX - touchObject.curX
  yDist = touchObject.startY - touchObject.curY
  r = Math.atan2(yDist, xDist)
  swipeAngle = Math.round((r * 180) / Math.PI)
  if (swipeAngle < 0) {
    swipeAngle = 360 - Math.abs(swipeAngle)
  }
  if ((swipeAngle <= 45 && swipeAngle >= 0) || (swipeAngle <= 360 && swipeAngle >= 315)) {
    return 'left'
  }
  if (swipeAngle >= 135 && swipeAngle <= 225) {
    return 'right'
  }

  return 'vertical'
}

export const swipeStart = (e: any): { dragging: boolean; touchObject: any } => {
  return {
    dragging: true,
    touchObject: {
      startX: e.touches ? e.touches[0].pageX : e.clientX,
      startY: e.touches ? e.touches[0].pageY : e.clientY,
      curX: e.touches ? e.touches[0].pageX : e.clientX,
      curY: e.touches ? e.touches[0].pageY : e.clientY,
    },
  }
}

export const swipeMove = <T extends {}>(e: any, spec: any): Partial<T> | undefined => {
  // spec also contains, trackRef and slideIndex
  const { animating, touchObject, scrolling } = spec
  if (scrolling) return
  if (animating) return e.preventDefault()
  let state: any = {}
  touchObject.curX = e.touches ? e.touches[0].pageX : e.clientX
  touchObject.curY = e.touches ? e.touches[0].pageY : e.clientY
  touchObject.swipeLength = Math.round(Math.sqrt(Math.pow(touchObject.curX - touchObject.startX, 2)))

  let positionOffset = touchObject.curX > touchObject.startX ? 1 : -1
  let touchSwipeLength = touchObject.swipeLength

  const swipeLeft = touchSwipeLength * positionOffset

  state = {
    ...state,
    touchObject,
    swipeLeft,
  }
  if (Math.abs(touchObject.curX - touchObject.startX) < Math.abs(touchObject.curY - touchObject.startY) * 0.8) {
    return state
  }

  if (touchObject.swipeLength > 10) {
    state['swiping'] = true
  }

  return state
}

export const swipeEnd = <T extends {}>(e: any, spec: any): Partial<T> => {
  const { dragging, touchObject, touchThreshold, onSwipe } = spec
  if (!dragging) {
    e.preventDefault()
    return {}
  }

  let minSwipe = touchThreshold
  let swipeDirection = getSwipeDirection(touchObject)

  // reset the state of touch related state variables.
  let state: any = {
    dragging: false,
    swiping: false,
    swiped: false,
    swipeLeft: null,
    touchObject: {},
  }

  if (!touchObject.swipeLength) {
    return state
  }

  if (touchObject.swipeLength > minSwipe && onSwipe) {
    onSwipe(swipeDirection)
  }
  return state
}
