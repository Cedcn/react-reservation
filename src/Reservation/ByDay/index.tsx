/** @jsx jsx */
import { jsx } from '@emotion/core'
import moment, { Moment } from 'moment'
import { map } from 'lodash'
import React from 'react'
import { getDateByArea, getNow } from '../utils'
import { Days, Offset, isSpecifiedDays } from '../interface'
import Calendar from '../components/Calendar'

interface ByDayProps {
  prefixCls?: string
  days?: Days
  area?: Offset
}

const ByDay: React.FC<ByDayProps> = (props) => {
  const { prefixCls = 'rV', area, days, ...rest } = props

  let startDay: Moment | null | undefined
  let endDay: Moment | null | undefined
  let disabledWeeks
  let disabledDays
  let specifiedDays

  if (isSpecifiedDays(days)) {
    startDay = moment.min(days)
    endDay = moment.max(days)
    specifiedDays = map(days, (day) => day.clone().startOf('day'))
  } else {
    startDay = days?.startDay
    endDay = days?.endDay
    disabledWeeks = days?.disabledWeeks
    disabledDays = days?.disabledDays
  }

  const today = getNow()
  const startMonthDay = startDay && startDay.isAfter(today, 'minute') ? startDay.startOf('day') : today.startOf('day')
  let endMonthDay = endDay ? endDay.endOf('day') : undefined
  if (area) {
    const endOffsetMonthDay = getDateByArea(area)

    if (!endMonthDay || (endMonthDay && endMonthDay.isAfter(endOffsetMonthDay, 'minute'))) {
      endMonthDay = endOffsetMonthDay
    }
  }

  return (
    <Calendar
      {...rest}
      className={`${prefixCls}-by-day`}
      prefixCls={prefixCls}
      specifiedDays={specifiedDays}
      disabledDays={disabledDays}
      disabledWeeks={disabledWeeks}
      startDay={startMonthDay}
      endDay={endMonthDay}
    />
  )
}

export default ByDay
