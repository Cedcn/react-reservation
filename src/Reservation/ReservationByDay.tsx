/** @jsx jsx */
import { jsx } from '@emotion/core'
import { ThemeProvider } from 'emotion-theming'
import { Moment } from 'moment'
import React, { useState, useEffect } from 'react'
import ByDay from './ByDay'
import { isEqual } from 'lodash'
import { Theme, Days, ByDayQuota, ByDayCellProps, Offset } from './interface'

export type Value = Moment | Moment[] | null

const defaultTheme = { borderColor: '#eee' }
export interface ReservationByDayProps {
  theme?: Theme
  prefixCls?: string
  defaultValue?: Value
  value?: Value
  onChange?: (value?: Value) => void
  days?: Days
  quotas?: ByDayQuota[] | ((startDay: Moment, endDay: Moment) => ByDayQuota[])
  advance?: Offset | boolean
  isMultiple?: boolean
  area?: Offset
  cellRenderer?: React.ComponentType<ByDayCellProps>
}

const ReservationByDay: React.FC<ReservationByDayProps> = (props) => {
  const { theme = defaultTheme, ...rest } = props

  const v = typeof props.value === 'undefined' ? props.defaultValue : props.value
  const [value, setValue] = useState<Value | undefined>(v)

  const onChange = (value?: Value) => {
    setValue(value)
    props.onChange && props.onChange(value)
  }

  useEffect(() => {
    if (!isEqual(props.value, value)) {
      setValue(props.value)
    }
  }, [props.value])

  return (
    <ThemeProvider theme={theme}>
      <ByDay value={value} onChange={onChange} {...rest} />
    </ThemeProvider>
  )
}

export default ReservationByDay
