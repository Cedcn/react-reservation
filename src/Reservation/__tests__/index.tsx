import React from 'react'
import Reservation from '../'
import renderer from 'react-test-renderer'
import MockDate from 'mockdate'

MockDate.set('2020-11-22')

it('renders correctly', () => {
  const tree = renderer.create(<Reservation />).toJSON()
  expect(tree).toMatchSnapshot()
})
