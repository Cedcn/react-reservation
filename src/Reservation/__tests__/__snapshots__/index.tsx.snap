// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h2>
    Repeat 
  </h2>,
  <div
    className="rV css-1dhoqxp-ReservationCalendar"
  >
    <div
      className="rV-header css-uyiikg-CalendarHeader"
    >
      <div
        className="rV-header__week"
      >
        <span>
          2019年12月
        </span>
      </div>
      <a
        className="rV-header__arrow rV-header__next-arrow css-1agcduh-CalendarHeader"
        onClick={[Function]}
        title="下一月"
      >
        下一月
      </a>
    </div>
    <div
      className="rV-table css-li9qyy-CalendarTable"
    >
      <div
        className="rV-thead css-2z8o9q-CalendarTHead"
      >
        <div
          className="rV-tr css-11qeklm-CalendarTHead"
          role="row"
        >
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sun"
          >
            <span
              className="rV-column-header-inner"
            >
              Sun
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Mon"
          >
            <span
              className="rV-column-header-inner"
            >
              Mon
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Tue"
          >
            <span
              className="rV-column-header-inner"
            >
              Tue
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Wed"
          >
            <span
              className="rV-column-header-inner"
            >
              Wed
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Thu"
          >
            <span
              className="rV-column-header-inner"
            >
              Thu
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Fri"
          >
            <span
              className="rV-column-header-inner"
            >
              Fri
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sat"
          >
            <span
              className="rV-column-header-inner"
            >
              Sat
            </span>
          </div>
        </div>
      </div>
      <div
        className="rV-viewer css-slbbi-CalendarTable"
      />
    </div>
  </div>,
  <h2>
    Repeat, set disabled weeks and set disabled days
  </h2>,
  <div
    className="rV css-1dhoqxp-ReservationCalendar"
  >
    <div
      className="rV-header css-uyiikg-CalendarHeader"
    >
      <div
        className="rV-header__week"
      >
        <span>
          2019年12月
        </span>
      </div>
      <a
        className="rV-header__arrow rV-header__next-arrow css-1agcduh-CalendarHeader"
        onClick={[Function]}
        title="下一月"
      >
        下一月
      </a>
    </div>
    <div
      className="rV-table css-li9qyy-CalendarTable"
    >
      <div
        className="rV-thead css-2z8o9q-CalendarTHead"
      >
        <div
          className="rV-tr css-11qeklm-CalendarTHead"
          role="row"
        >
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sun"
          >
            <span
              className="rV-column-header-inner"
            >
              Sun
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Mon"
          >
            <span
              className="rV-column-header-inner"
            >
              Mon
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Tue"
          >
            <span
              className="rV-column-header-inner"
            >
              Tue
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Wed"
          >
            <span
              className="rV-column-header-inner"
            >
              Wed
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Thu"
          >
            <span
              className="rV-column-header-inner"
            >
              Thu
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Fri"
          >
            <span
              className="rV-column-header-inner"
            >
              Fri
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sat"
          >
            <span
              className="rV-column-header-inner"
            >
              Sat
            </span>
          </div>
        </div>
      </div>
      <div
        className="rV-viewer css-slbbi-CalendarTable"
      />
    </div>
  </div>,
  <h2>
    Repeat, set start day and set end day
  </h2>,
  <div
    className="rV css-1dhoqxp-ReservationCalendar"
  >
    <div
      className="rV-header css-uyiikg-CalendarHeader"
    >
      <div
        className="rV-header__week"
      >
        <span>
          2020年02月
        </span>
      </div>
      <a
        className="rV-header__arrow rV-header__next-arrow css-1agcduh-CalendarHeader"
        onClick={[Function]}
        title="下一月"
      >
        下一月
      </a>
    </div>
    <div
      className="rV-table css-li9qyy-CalendarTable"
    >
      <div
        className="rV-thead css-2z8o9q-CalendarTHead"
      >
        <div
          className="rV-tr css-11qeklm-CalendarTHead"
          role="row"
        >
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sun"
          >
            <span
              className="rV-column-header-inner"
            >
              Sun
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Mon"
          >
            <span
              className="rV-column-header-inner"
            >
              Mon
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Tue"
          >
            <span
              className="rV-column-header-inner"
            >
              Tue
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Wed"
          >
            <span
              className="rV-column-header-inner"
            >
              Wed
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Thu"
          >
            <span
              className="rV-column-header-inner"
            >
              Thu
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Fri"
          >
            <span
              className="rV-column-header-inner"
            >
              Fri
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sat"
          >
            <span
              className="rV-column-header-inner"
            >
              Sat
            </span>
          </div>
        </div>
      </div>
      <div
        className="rV-viewer css-slbbi-CalendarTable"
      />
    </div>
  </div>,
  <h2>
    Specified days
  </h2>,
  <div
    className="rV css-1dhoqxp-ReservationCalendar"
  >
    <div
      className="rV-header css-uyiikg-CalendarHeader"
    >
      <div
        className="rV-header__week"
      >
        <span>
          2020年02月
        </span>
      </div>
      <a
        className="rV-header__arrow rV-header__next-arrow css-1agcduh-CalendarHeader"
        onClick={[Function]}
        title="下一月"
      >
        下一月
      </a>
    </div>
    <div
      className="rV-table css-li9qyy-CalendarTable"
    >
      <div
        className="rV-thead css-2z8o9q-CalendarTHead"
      >
        <div
          className="rV-tr css-11qeklm-CalendarTHead"
          role="row"
        >
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sun"
          >
            <span
              className="rV-column-header-inner"
            >
              Sun
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Mon"
          >
            <span
              className="rV-column-header-inner"
            >
              Mon
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Tue"
          >
            <span
              className="rV-column-header-inner"
            >
              Tue
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Wed"
          >
            <span
              className="rV-column-header-inner"
            >
              Wed
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Thu"
          >
            <span
              className="rV-column-header-inner"
            >
              Thu
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Fri"
          >
            <span
              className="rV-column-header-inner"
            >
              Fri
            </span>
          </div>
          <div
            className="rV-th css-rbejd6"
            role="columnheader"
            title="Sat"
          >
            <span
              className="rV-column-header-inner"
            >
              Sat
            </span>
          </div>
        </div>
      </div>
      <div
        className="rV-viewer css-slbbi-CalendarTable"
      />
    </div>
  </div>,
  <h2>
    Time Buckets
  </h2>,
  <div
    className="rT css-1xd6wqv-ReservationTimeBuckets"
  >
    <div
      className="rT-header css-lxofq8-TimeBucketsHeader"
    >
      <div
        className="rT-header__week"
      >
        February 2, 2020
         - 
        February 8, 2020
      </div>
      <a
        className="rT-header__arrow rT-header__next-arrow css-k77o27-TimeBucketsHeader"
        onClick={[Function]}
        title="下一周"
      >
        下一周
      </a>
    </div>
    <div
      className="rT-table css-6w6r4c-TimeBucketsTable"
    >
      <div
        className="rT-thead css-avz3aa-TimeBucketsTable"
      >
        <div
          className="rT-tr is-current-week css-o90toh-TimeBucketsTable"
        >
          <div
            className="rT-th rT-column css-goa6kd-TimeBucketsTable"
            role="column"
          >
            <span>
              时段/日期
            </span>
          </div>
          <div
            className="rT-th css-rbejd6"
            title="Sun"
          >
            <span
              className="rT-th-inner"
            >
              <span>
                Sun
              </span>
              <span>
                2.2
              </span>
            </span>
          </div>
          <div
            className="rT-th css-rbejd6"
            title="Mon"
          >
            <span
              className="rT-th-inner"
            >
              <span>
                Mon
              </span>
              <span>
                2.3
              </span>
            </span>
          </div>
          <div
            className="rT-th css-rbejd6"
            title="Tue"
          >
            <span
              className="rT-th-inner"
            >
              <span>
                Tue
              </span>
              <span>
                2.4
              </span>
            </span>
          </div>
          <div
            className="rT-th css-rbejd6"
            title="Wed"
          >
            <span
              className="rT-th-inner"
            >
              <span>
                Wed
              </span>
              <span>
                2.5
              </span>
            </span>
          </div>
          <div
            className="rT-th css-rbejd6"
            title="Thu"
          >
            <span
              className="rT-th-inner"
            >
              <span>
                Thu
              </span>
              <span>
                2.6
              </span>
            </span>
          </div>
          <div
            className="rT-th css-rbejd6"
            title="Fri"
          >
            <span
              className="rT-th-inner"
            >
              <span>
                Fri
              </span>
              <span>
                2.7
              </span>
            </span>
          </div>
          <div
            className="rT-th css-rbejd6"
            title="Sat"
          >
            <span
              className="rT-th-inner"
            >
              <span>
                Sat
              </span>
              <span>
                2.8
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        className="rT-viewer rT-viewer--weeks css-1y9ig7-TimeBucketsTable"
      />
    </div>
  </div>,
  <h2>
    Time Buckets list
  </h2>,
  <div
    className="rT css-1xd6wqv-ReservationTimeBuckets"
  >
    <div
      className="rT-header css-lxofq8-TimeBucketsHeader"
    >
      <div
        className="rT-header__week"
      >
        February 2, 2020
         - 
        February 8, 2020
      </div>
      <a
        className="rT-header__arrow rT-header__next-arrow css-k77o27-TimeBucketsHeader"
        onClick={[Function]}
        title="下一周"
      >
        下一周
      </a>
    </div>
    <div
      className="rT-viewer--tabs css-zxfxe-TimeBucketsTabs"
    >
      <br />
    </div>
  </div>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
  <p>
    12313123
  </p>,
]
`;
